pkgs/servers/xandikos/default.nix

```diff
@@ -1,38 +1,71 @@
 { lib
+, buildPythonApplication
 , fetchFromGitHub
-, python3Packages
 , nixosTests
+, pythonOlder
+# dependencies
+, aiohttp
+, aiohttp-openmetrics
+, defusedxml
+, dulwich
+, icalendar
+, jinja2
+, multidict
+, vobject
+# build dependencies
+, setuptools
+, wheel
+, pytestCheckHook
 }:
 
-python3Packages.buildPythonApplication rec {
+buildPythonApplication rec {
   pname = "xandikos";
-  version = "0.2.8";
+  version = "0.2.10";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "jelmer";
     repo = "xandikos";
     rev = "v${version}";
-    sha256 = "sha256-KDDk0QSOjwivJFz3vLk+g4vZMlSuX2FiOgHJfDJkpwg=";
+    hash = "sha256-SqU/K3b8OML3PvFmP7L5R3Ub9vbW66xRpf79mgFZPfc=";
   };
 
-  propagatedBuildInputs = with python3Packages; [
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = [
     aiohttp
+    aiohttp-openmetrics
     dulwich
     defusedxml
     icalendar
     jinja2
     multidict
-    prometheus-client
+    vobject
   ];
 
   passthru.tests.xandikos = nixosTests.xandikos;
 
-  nativeCheckInputs = with python3Packages; [ pytestCheckHook ];
+  nativeCheckInputs = [ pytestCheckHook ];
   disabledTests = [
     # these tests are failing due to the following error:
-    # TypeError: expected str, bytes or os.PathLike object, not int
-    "test_iter_with_etag"
-    "test_iter_with_etag_missing_uid"
+    # TypeError: index_entry_from_stat() missing 1 required positional argument
```

Comment:
I think something is wrong.
Do these tests also fail upstream?

Reply:
I disabled those before I saw dulwich needed updating. With dulwich 0.21.6 its no longer necessary.